stdlib/SparseArrays/src/linalg.jl

```diff
@@ -1,12 +1,35 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
-import LinearAlgebra: checksquare
+import LinearAlgebra: checksquare, sym_uplo
 using Random: rand!
 
 # In matrix-vector multiplication, the correct orientation of the vector is assumed.
 const StridedOrTriangularMatrix{T} = Union{StridedMatrix{T}, LowerTriangular{T}, UnitLowerTriangular{T}, UpperTriangular{T}, UnitUpperTriangular{T}}
 const AdjOrTransStridedOrTriangularMatrix{T} = Union{StridedOrTriangularMatrix{T},Adjoint{<:Any,<:StridedOrTriangularMatrix{T}},Transpose{<:Any,<:StridedOrTriangularMatrix{T}}}
 
+for (Wrapper, operation) ∈ [(:Hermitian, :adjoint), (:Symmetric, :transpose)]
+    @eval begin
+        (+)(A::SparseMatrixCSC, B::$Wrapper{<:Any,<:AbstractSparseMatrix}) = A + sparse(B)
+        (+)(A::$Wrapper{<:Any,<:AbstractSparseMatrix}, B::SparseMatrixCSC) = B + A
+
+        (+)(A::SparseMatrixCSC, B::$Wrapper) = A + collect(B)
+        (+)(A::$Wrapper, B::SparseMatrixCSC) = B + A
+
+        function +(A::$Wrapper{<:Any, <:SparseMatrixCSC}, B::$Wrapper{<:Any, <:SparseMatrixCSC})
+            if A.uplo == B.uplo
+                $Wrapper(parent(A) + parent(B), sym_uplo(A.uplo))
+            elseif A.uplo == 'U'
+                $Wrapper(parent(A) + $operation(parent(B)), :U)
+            else
+                $Wrapper($operation(parent(A)) + parent(B), :U)
+            end
+        end
```

Comment:
As I said [here](https://github.com/JuliaLang/julia/pull/35325#discussion_r407100805), this should go to LinearAlgebra/src/symmetric.jl around lines 465, but without the `<:SparseMatrixCSC` restriction. First, you need to do that anyway to avoid the ambiguity, and second, this approach here is even beneficial for dense matrices, compared to the code there.

Reply:
BTW, the (partially) real versions there,

https://github.com/JuliaLang/julia/blob/0c284839fef6c8c153edc01fddfa37a9f5ac6752/stdlib/LinearAlgebra/src/symmetric.jl#L467-L468

would also benefit from this more careful treatment.

Reply:
Hm, I'm now not really seeing the advantage of that big `if ... elseif ... else` block over the way it's done here in `LinearAlgebra/src/symmetric.jl`. I think as long as I properly handle `::Sparse + ::Hermitian` and such, the implementations of `symmetric.jl` seem like they should do the right thing.

Reply:
Did you try my benchmarks here https://github.com/JuliaLang/julia/pull/35325#discussion_r407100805? The advantage is a significant speedup.

Reply:
Oh, and the other advantage is that it actually works... and doesn't lead to ambiguity errors.

Reply:
One more: I thought we're discussing here the `Symmetric + Symmetric` and `Hermitian + Hermitian` case. With the most generic methods given in `LinearAlgebra`, I don't see how you avoid the method ambiguity.

Reply:
Right, sorry about that I see now.

Reply:
I really appreciate your patient and persistent help with this PR by the way!